src/mgr/MDSPerfMetricTypes.h

```diff
@@ -317,6 +317,7 @@ std::ostream &operator<<(std::ostream &os, const MDSPerfMetricQuery &query);
 struct MDSPerfCollector : PerfCollector {
   std::map<MDSPerfMetricKey, PerformanceCounters> counters;
   std::set<mds_rank_t> delayed_ranks;
+  utime_t last_updated;
```

Comment:
We could use monotonic clock. But it would prevent the dumping, as mgr/PyFormatter doesn't support conversions from monotonic clock. No such issues with `utime_t`.

Please check: 
https://github.com/ceph/ceph/blob/8effd036db9c49d55875a5109c51b7497b99db35/src/mgr/ActivePyModules.cc#L1443
https://github.com/ceph/ceph/blob/master/src/mgr/PyFormatter.h#L89-L95
https://github.com/ceph/ceph/blob/8effd036db9c49d55875a5109c51b7497b99db35/src/pybind/mgr/stats/fs/perf_stats.py#L319

Reply:
I would like to use utime_t here, analyzing the complexity the suggested change would make on the mgr side and the time conversion operations in perf_stats.py. The other comments are fixed.

Reply:
Monotonic clocks are preferred over realtime clocks for the very reason that it (monotonic) clocks can't go backwards. That's essential for things like such.